tests/qa_automation/acceptance.pm

```diff
@@ -65,8 +65,25 @@ sub run() {
     # QA DB upload happens for each module
     # output the failed tests to serial console
     type_string "export PS1=#\n";
+    # qa_testset_automation deleted oldlog folders, so we add it back to adopt it
+    assert_script_run "mkdir /var/log/qa/oldlogs; cd /var/log/qa/oldlogs; for i in /var/log/qaset/log/*.bz2; do tar -xjvf \$i; done";
     type_string "find /var/log/qa/oldlogs/ -name test_results | xargs grep -l -B1 ^1 | tee -a /dev/$serialdev\n";
     assert_screen 'no_output_from_qa_find';
+    # create a junit report
```

Comment:
I don't like this loop at all. All that's different between the ifs is argument to -t, so better calculate that variable in an if-else and then have one line to run the command using the variable.